client/src/components/applicants/Applicants.js

```diff
@@ -35,10 +36,12 @@ export class Applicants extends Component {
     return (
       <div className='applicants'>
         <div className='applicants__title'>{jobPosting.title}</div>
-        <button className='applicants__button'>
-          Copy as a template
-        </button>
-        <div className='applicantion-stages'>
+        <Link to='/jobposting/new' style={{ textDecoration: 'none' }}>
+          <button className='applicants__button' onClick={() => this.props.copyStages(stages)}>
```

Comment:
Muuten näyttää hyvälle, mutta tän onClick -kutsun vois muokata vähän toiseen muotoon. Jos käyttää renderissä bindia tai arrow funkkaria, niin joka kerta kun tuo komponentti renderöidään luodaan uusi funktio. Meidän kokoisessa projektissa sillä nyt ei välttämättä ole toistaiseksi väliä, mutta ihan hyvä koodin ylläpidettävyyden kannalta kuitenkin käyttää tässäkin parhaita tapoja.

Eli tässä pitäisi tehdä joko:
```
handleCopyStages() {
    const { stages, copyStages } = this.props
    copyStages(stages)
  }
```
Ja sitten sitoa (bind) konstruktorissa `this.handleCopyStages = this.handleCopyStages.bind(this)`

**Tai** jos ei halua tehdä noita bindeja, niin kutsuu vain arrow funktiota (tätä tapaa mie yleensä käytän):
```
handleCopyStages = () => {
    const { stages, copyStages } = this.props
    copyStages(stages)
  }
```

Ja molempien tapauksien jälkeen renderissä kutsuu vaan onClick={this.handleCopyStages}. Suosittelen vilkaisemaan https://reactjs.org/docs/faq-functions.html, siellä on aika hyvin selitetty auki mitkä tavat on parhaat ja miksi.